README.md

```diff
@@ -123,6 +123,26 @@ let x = 0.0
 end
 ```
 
+Note that the let block is used here to create a scope to allow assignment of 
+`x` within `@byrow!`.
+
+`byrow!` also supports special syntax for allocating new columns to make
+`byrow!` more useful for data transformations. The syntax `@newcol
+x::Array{Int}` allocates a new column `:x` with an `Array` container with eltype
+`Int`. Note that the returned AbstractDataFrame includes these new columns, but
```

Comment:
Even though `setindex!` may not be allowed in type-stable AbstractDataFrames, macros have enough magic to still allow re-assignment. We can have `@assign d[:a] = newCol` do the following:

```julia
df = @transform(df, a = newCol)
```
We can take the same approach with `@newcol`/`byrow!`. However, all of the other DataFramesMeta functions require the use of the returned value, so maybe the current approach is more consistent. I'm still undecided here.